compiler/rustc_hir_analysis/src/check/method/suggest.rs

```diff
@@ -1201,6 +1203,71 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    fn suggest_range_for_iter(
+        &self,
+        tcx: TyCtxt<'tcx>,
+        actual: Ty<'tcx>,
+        source: SelfSource<'tcx>,
+        span: Span,
+        item_name: Ident,
+        ty_str: &str,
+    ) -> bool {
+        if let SelfSource::MethodCall(expr) = source {
+            for (_, parent) in tcx.hir().parent_iter(expr.hir_id) {
+                if let Node::Expr(parent_expr) = parent && is_range_literal(parent_expr) {
+                    let span_included = match parent_expr.kind {
+                            hir::ExprKind::Struct(_, eps, _) => eps.iter().any(|ep| ep.expr.span.contains(span)),
```

Comment:
I think this `span_included` is no need, since parent must contains the sub-expr? @lcnr

Reply:
emm, we need to check it's the right bound of a range.